### PR TITLE
[Executor] Add e2e tests for including openai tokens in otel traces

### PR DESCRIPTION
**Pull request by lumoslnt**
# Description

Add e2e tests for including openai tokens in otel traces.

# All Promptflow Contribution checklist:
- [x] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow/tests/executor/e2etests/test_traces.py [hunk 1]
```diff
@@ -289,11 +320,14 @@ def assert_otel_traces(self, flow_file):
             assert isinstance(span.name, str)
             assert span.attributes["line_run_id"] == line_run_id
             assert span.attributes["framework"] == "promptflow"
-            expected_span_type = TraceType.FUNCTION
             if span.parent is None:
```
Comment (thy09) [inline src/promptflow/tests/executor/e2etests/test_traces.py:323]: Maybe we could extract a function for this?

File: src/promptflow/tests/executor/e2etests/test_traces.py [hunk 2]
```diff
@@ -302,6 +336,31 @@ def assert_otel_traces(self, flow_file):
             output = json.loads(span.attributes["output"])
             assert isinstance(inputs, dict)
             assert output is not None
+        self.validate_openai_tokens(span_list)
+
+    def validate_openai_tokens(self, span_list):
```
Comment (thy09) [inline src/promptflow/tests/executor/e2etests/test_traces.py:341]: Add some comments to explain what we are validating.

File: src/promptflow/tests/test_configs/flows/flow_with_async_llm_tasks/chat.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+def create_messages(question, chat_history):
+    yield {"role": "system", "content": "You are a helpful assistant."}
+    for chat in chat_history:
+        yield {"role": "user", "content": chat["inputs"]["question"]}
+        yield {"role": "assistant", "content": chat["outputs"]["answer"]}
+        yield {"role": "user", "content": question}
+
+@trace
+async def chat(connection: AzureOpenAIConnection, question: str, chat_history: List, model: str) -> str:
+    if IS_LEGACY_OPENAI:
```
Comment (thy09) [inline src/promptflow/tests/test_configs/flows/flow_with_async_llm_tasks/chat.py:38]: When will we test this code path?

File: src/promptflow/tests/test_configs/flows/flow_with_async_llm_tasks/chat.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+            messages=list(create_messages(question, chat_history)),
+            temperature=1.0,
+            top_p=1.0,
+            n=1,
+            stop=None,
+            max_tokens=16,
+            **dict(connection),
+        )
+    else:
+        completion = get_client(connection).chat.completions.create(
```
Comment (thy09) [inline src/promptflow/tests/test_configs/flows/flow_with_async_llm_tasks/chat.py:50]: We should use async client and await to get the values.

Comment (lumoslnt) [reply src/promptflow/tests/test_configs/flows/flow_with_async_llm_tasks/chat.py]: Refined.

File: src/promptflow/tests/test_configs/flows/flow_with_async_llm_tasks/chat.py (hunk 3 above)
Comment (lumoslnt) [reply src/promptflow/tests/test_configs/flows/flow_with_async_llm_tasks/chat.py]: Removed the old path.

File: src/promptflow/tests/executor/e2etests/test_traces.py (hunk 2 above)
Comment (lumoslnt) [reply src/promptflow/tests/executor/e2etests/test_traces.py:348]: Added.

File: src/promptflow/tests/executor/e2etests/test_traces.py (hunk 1 above)
Comment (lumoslnt) [reply src/promptflow/tests/executor/e2etests/test_traces.py]: Done.

**Review by thy09 — APPROVED** [review]

**Review by zzn2 — APPROVED** [review]

**Pull request merged**